### PR TITLE
Update docs for 1.3

### PR DESCRIPTION
**Pull request by sgrif**
This adds a guide covering `diesel.toml`, and replaces all mentions of
`infer_schema!` with `diesel print-schema`.

### CODE REVIEW

File: source/views/guides/configuring-diesel-cli.html.slim [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    .banner-wrapper
+      h2.banner__heading Configuring Diesel CLI
+
+main
+  section.demo
+    .content-wrapper
+      .guide-wrapper
+        markdown:
+          Diesel CLI is an optional tool Diesel provides to manage your
+          database schema. It's main two roles are to run database
```
Comment (Diggsey) [inline source/views/guides/configuring-diesel-cli.html.slim]: Its

File: source/views/guides/configuring-diesel-cli.html.slim [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          migrations, and create a Rust file which represents your
```
Comment (Diggsey) [inline source/views/guides/configuring-diesel-cli.html.slim]: and to create

File: source/views/guides/configuring-diesel-cli.html.slim [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          database schema.
+
+          The behavior of Diesel CLI can be configured through a toml file. By
+          default Diesel will look for `diesel.toml` in the same directory as
+          your `Cargo.toml` file. You can provide a different config file by
+          setting the `DIESEL_CONFIG_FILE` environment variable, or passing
+          `--config-file` on the command line. You can get a basic config file
+          with some defaults provided by running `diesel setup`.
```
Comment (Diggsey) [inline source/views/guides/configuring-diesel-cli.html.slim]: What will `diesel setup` do on a pre-existing project?

File: source/views/guides/configuring-diesel-cli.html.slim [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+              code
+                | [print-schema]
+                  # Add types from `diesel_full_text_search` like `tsvector`
+                  import_types = ["diesel::sql_types::*", "diesel_full_text_search::types::*"]
+
+        markdown:
+          The `patch_file` field
+          ------------------------
+
+          Specifies a `.patch` file to be applied to your schema after it is
```
Comment (Diggsey) [inline source/views/guides/configuring-diesel-cli.html.slim]: Is the schema generation 100% deterministic, especially around ordering?

File: source/views/guides/configuring-diesel-cli.html.slim (hunk 3 above)
Comment (sgrif) [reply source/views/guides/configuring-diesel-cli.html.slim]: - Create the migrations directory if it doesn't already exist
- Generate the `000000` migration if it doesn't already exist and PG is being used
- Generate `diesel.toml` if it doesn't already exist
- `diesel database setup`
  - Create the database if it doesn't exist
  - Run migrations

It's designed to be idempotent

File: source/views/guides/configuring-diesel-cli.html.slim (hunk 4 above)
Comment (sgrif) [reply source/views/guides/configuring-diesel-cli.html.slim]: Yes. Tables are ordered alphabetically. Columns are ordered by definition order.

File: source/views/guides/configuring-diesel-cli.html.slim (hunk 3 above)
Comment (Diggsey) [reply source/views/guides/configuring-diesel-cli.html.slim]: That's good, do you think it's worth adding a note here that it won't overwrite existing files?

Comment (sgrif) [reply source/views/guides/configuring-diesel-cli.html.slim]: Seems reasonable

**Pull request merged**